### DISCUSSION TITLE
Error from messaging.go but I am not trying to publish

### DISCUSSION THREAD

**Discussion by herrdragon** [edgexfoundry/community#59] [Q&A] [answered]
Hi Team,

I am getting this error: source=messaging.go:231 msg="MessageBus trigger: Could not publish to topic '' for pipeline 'Enrich': Missing configuration 'PublishHostInfo' : Unable to create a connection for publishing"
I am subscribed to a topic but I am not trying to publish anything but it seems somewhere in the code is trying to do so but not sure why . Any ideas? Thank you

**Comment by lenny-goodell** [comment]
Please provide the version of EdgeX you are using.

**Reply by lenny-goodell** [reply]
BTW, if you set ResponseData it will attempt to publish. In your case you don't have the PublishHostInfo in your configuration. If you don't want to publish, don't set the ResponseData.

https://docs.edgexfoundry.org/2.1/microservices/application/AppFunctionContextAPI/#responsedata
https://docs.edgexfoundry.org/2.1/microservices/application/BuiltIn/#set-response-data
https://docs.edgexfoundry.org/2.1/microservices/application/Triggers/#messagebus-connection-configuration

**Reply by herrdragon** [reply]
Hi Lenny,

I am running version 2.1 and I am not setting ResponseData. I have removed the PublishHostInfo to see if that would prevent from trying to publish but I still get the error.

**Comment by lenny-goodell** [comment, marked as answer]
What application service are you running? Custom or app-service-configurable?

Not the error is because `PublishHostInfo` is not set when it goes to publish.

Not from the code the only way it sets to line 231 is if the `ResponseData` is set.
https://github.com/edgexfoundry/app-functions-sdk-go/blob/v2.1.0/internal/trigger/messagebus/messaging.go#L200-L236

**Comment by herrdragon** [comment]
Perfect! Since is checking for ResponseData != nil, I set ResponseData to nil and now I don't have the error. Thank you for your help!
